sound/soc/sof/Kconfig

```diff
@@ -37,14 +37,25 @@ config SND_SOC_SOF_ACPI
 config SND_SOC_SOF_ACPI_DEV
 	tristate
 
-config SND_SOC_SOF_OF
-	tristate "SOF OF enumeration support"
+config SND_SOC_SOF_OF_ENABLE
+	bool "Enable SOF OF enumeration support"
 	depends on OF || COMPILE_TEST
 	help
-	  This adds support for Device Tree enumeration. This option is
+	  This enables support for Device Tree enumeration. This option is
 	  required to enable i.MX8 or Mediatek devices.
 	  Say Y if you need this option. If unsure select "N".
 
+config SND_SOC_SOF_OF
+	tristate
+	depends on SND_SOC_SOF
+	depends on OF || COMPILE_TEST
+	help
+	  This is the real option to add Device Tree enumeration
+	  implementation. The SND_SOC_SOF_OF_ENABLE is there only let
+	  Device Tree depended platforms' top levevel menus to depend
+	  on it without a circular dependency. This option is sleceted
+	  by the platform menu items needing it.
+
```

Comment:
I don't follow this.

````
config SND_SOC_SOF_IMX_TOPLEVEL
	bool "SOF support for NXP i.MX audio DSPs"
	depends on ARM64|| COMPILE_TEST
	depends on SND_SOC_SOF_OF

config SND_SOC_SOF_IMX_COMMON
	tristate
	select SND_SOC_SOF_OF_DEV
	select SND_SOC_SOF
````

something's off here. In addition we have the same solution for ACPI, PCI and OF, so I am struggling to see why only OF has a problem.

Reply:
Hmmm, it indeed looks like I've made a mess with this commit. It looks like the problem that I am trying to solve here is my own doing, exotic COMPILE_TEST configuration together with one wrongly placed "depends on".

I'll drop the last commit. Oh well, at least I got couple of hours practice on Kconfig issues.

Reply:
No worries @jsarha it's hard to get things right, I don't know how many times we've had to revisit the Kconfig because of a dependency we didn't handle well. I personally broke all this more than once :-)